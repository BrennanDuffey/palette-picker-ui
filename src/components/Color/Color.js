function Color(props) {
  return (
    <section>
      <div style=`background-color:${props.hexcode};`></div>
      {props.isLocked && <p onClick={props.setColor}> LOCKED </p>}
      {!props.isLocked && <p onClick={props.setColor}> UNLOCKED </p>}
      <p> {props.hexcode} </p>
    </section>
  )
}