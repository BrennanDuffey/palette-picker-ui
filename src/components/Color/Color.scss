.Color-block {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
}

.color {
  width: 100%;
  height: 400px;
}

.fa-lock,
.fa-unlock {
  color: $gray;
  cursor: pointer;
  margin: 10px 0;

  &:hover {
    color: black;
  }
}

.fa-lock {
  color: black;
}

.hex-code {
  font-size: 1.1em;
  font-weight: 600;
  text-transform: uppercase;
}