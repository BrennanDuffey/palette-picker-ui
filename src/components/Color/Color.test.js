import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Color toggleLocked={jest.fn()} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call toggleLocked on click', () => {
    wrapper.find('#locked').simulate('click');
    expect(wrapper.instance().toggleLocked).toHaveBeenCalled();
  });
});