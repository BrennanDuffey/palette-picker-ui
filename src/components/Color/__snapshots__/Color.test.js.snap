// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color should match snapshot 1`] = `
<section
  className="palette"
>
  <div
    className="color"
    style={
      Object {
        "background": undefined,
      }
    }
  />
  <p
    onClick={[Function]}
  >
     UNLOCKED 
  </p>
  <p>
     
     
  </p>
</section>
`;
